homeassistant/components/system_bridge/__init__.py

```diff
@@ -0,0 +1,261 @@
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from .const import BRIDGE_CONNECTION_ERRORS, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["sensor"]
+
+CONF_ARGUMENTS = "arguments"
+CONF_BRIDGE = "bridge"
+CONF_WAIT = "wait"
+
+SERVICE_SEND_COMMAND = "send_command"
+SERVICE_SEND_COMMAND_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_BRIDGE): cv.string,
+        vol.Required(CONF_COMMAND): cv.string,
+        vol.Optional(CONF_ARGUMENTS, []): cv.string,
+    }
+)
+SERVICE_OPEN = "open"
+SERVICE_OPEN_SCHEMA = vol.Schema(
+    {vol.Required(CONF_BRIDGE): cv.string, vol.Required(CONF_PATH): cv.string}
+)
+
+
+async def async_setup(hass: HomeAssistantType, config: ConfigType) -> bool:
+    """Set up the System Bridge integration."""
+
+    async def handle_send_command(call):
+        """Handle the send_command service call."""
+        device_registry = await hass.helpers.device_registry.async_get_registry()
+        device_entry = device_registry.async_get(call.data.get(CONF_BRIDGE))
+
+        command = call.data.get(CONF_COMMAND)
+        arguments = shlex.split(call.data.get(CONF_ARGUMENTS, ""))
+
+        coordinator: DataUpdateCoordinator = hass.data[DOMAIN][
+            next(iter(device_entry.config_entries))
+        ]
+        bridge: Bridge = coordinator.data
+
+        _LOGGER.debug(
+            "Command payload: %s",
+            {CONF_COMMAND: command, CONF_ARGUMENTS: arguments, CONF_WAIT: False},
+        )
+        try:
+            response: CommandResponse = await bridge.async_send_command(
+                {CONF_COMMAND: command, CONF_ARGUMENTS: arguments, CONF_WAIT: False}
+            )
+            if response.success:
+                _LOGGER.debug(
+                    "Sent command. Response message was: %s", response.message
+                )
+            else:
+                _LOGGER.warning(
+                    "Error sending command. Response message was: %s", response.message
+                )
+        except (BridgeAuthenticationException, *BRIDGE_CONNECTION_ERRORS) as exception:
+            _LOGGER.warning("Error sending command. Error was: %s", exception)
+
+    async def handle_open(call):
+        """Handle the open service call."""
+        device_registry = await hass.helpers.device_registry.async_get_registry()
+        device_entry = device_registry.async_get(call.data.get(CONF_BRIDGE))
+
+        path = call.data.get(CONF_PATH)
+
+        coordinator: DataUpdateCoordinator = hass.data[DOMAIN][
+            next(iter(device_entry.config_entries))
+        ]
+        bridge: Bridge = coordinator.data
+
+        _LOGGER.debug("Open payload: %s", {CONF_PATH: path})
+        try:
+            await bridge.async_open({CONF_PATH: path})
+            _LOGGER.debug("Sent open request")
+        except (BridgeAuthenticationException, *BRIDGE_CONNECTION_ERRORS) as exception:
+            _LOGGER.warning("Error sending. Error was: %s", exception)
+
+    hass.services.async_register(
```

Comment:
Please move the service registration to `async_setup_entry`. Check if the service is registered with `hass.services.has_service` if more than one entry can be configured.